### PR TITLE
Fix pulp export creation task dispatching

### PR DESCRIPTION
**Pull request by mdellweg**
[noissue]

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/app/tasks/export.py [hunk 1]
```diff
@@ -156,14 +159,23 @@ def pulp_export(the_export):
     3) Compute and store the sha256 and filename of the resulting tar.gz/chunks
 
     Args:
-        the_export (models.PulpExport): PulpExport instance
+        exporter_pk (str): PulpExporter
+        params (dict): request data
+        validated_data (dict): serializer output
```
Comment (ggainey) [inline pulpcore/app/tasks/export.py:164]: I don't see validated_data in the args list

File: pulpcore/app/tasks/export.py [hunk 2]
```diff
@@ -156,14 +159,23 @@ def pulp_export(the_export):
...
 
     Raises:
         ValidationError: When path is not in the ALLOWED_EXPORT_PATHS setting,
             OR path exists and is not a directory
     """
+    pulp_exporter = PulpExporter.objects.get(pk=exporter_pk)
+    serializer = PulpExportSerializer(data=params, context={"exporter": pulp_exporter})
+    serializer.is_valid(raise_exception=True)
```
Comment (ggainey) [inline pulpcore/app/tasks/export.py:172]: With this change, we validate the data twice - once at the viewset, so we can return errors quickly, and then again in the task, so we can use the resulting validated data. I don't think this is a huge issue - but it does double the db-accesses for this path. Not sure there's anything to be done about it, given the constraints that drive this change though.

**Review by ggainey — CHANGES_REQUESTED** [review]
A minor nit and an observation, but generally looks good to me! Thanks for doing the work!

File: pulpcore/app/tasks/export.py (hunk 1 above)
Comment (mdellweg) [reply pulpcore/app/tasks/export.py:164]: right (it was there for a brief moment...)

File: pulpcore/app/tasks/export.py (hunk 2 above)
Comment (mdellweg) [reply pulpcore/app/tasks/export.py:172]: Yes. But my view is, that both validations can be separated by a significant amount of time. And the circumstances might even make once valid data invalid at the point the task is ready to start. So we really should validate it there.

Comment (ggainey) [reply pulpcore/app/tasks/export.py:171]: That's a good observation, concur.

**Review by ggainey — APPROVED** [review]

**Review by ipanova — APPROVED** [review]

**Review by daviddavis — APPROVED** [review]

**Pull request merged**